woocommerce/woocommerce
# Product variations - Set fixed height for card

### All Submissions:






### Changes proposed in this Pull Request:


This PR sets a fixed min-height for the product variation card.

The min-height will vary depending [on the rows displayed](https://github.com/woocommerce/woocommerce/issues/35789#issuecomment-1354615049).

| Rows | Space |
|---|---|
| rows count <= 5 | the space for 5 rows |
| 5 < rows count >= 10 | the space for 10 rows |
| 10 < rows count >= 25 | the space for 25 rows |

* There is no space saved for more than 25 rows and for the mobile view.

Closes #35789.






### How to test the changes in this Pull Request:



0. Right now, the pagination works for 25, 50, 75, and 100 rows, but here we will need 5, 10, and 25. So I created [this branch](https://github.com/woocommerce/woocommerce/tree/add/35789_set_variations_list_fixed_height_test) to tweak the logic to be able to test the pagination. If you test the PR as it is now, you only will see the pagination if you have more variations than the items set to be shown; by default it's 25.
1. Checkout the branch `add/35789_set_variations_list_fixed_height_test`.
2. Create a product with more than one but less than 5 variations.
3. Verify that there is an empty space after the variations.
4. Now select 50 rows per page and verify that there is no space between the variations and the pagination component. (Don't pay attention to the `Page 1 of 2`; that's expected in order to show the pagination).
5. Now select 25 rows per page and verify that there is space for 25 rows in total before the pagination component.



### Other information:




### FOR PR REVIEWER ONLY:
